python/lsst/pipe/tasks/hips.py

```diff
@@ -0,0 +1,545 @@
+
+        for sub in [parser_segment, parser_build]:
+            # These arguments are in common.
+            sub.add_argument(
+                "-b",
+                "--butler-config",
+                type=str,
+                help="Path to data repository or butler configuration.",
+                required=True,
+            )
+            sub.add_argument(
+                "-p",
+                "--pipeline",
+                type=str,
+                help="Pipeline file, limited to one task.",
+                required=True,
+            )
+            sub.add_argument(
+                "-i",
+                "--input",
+                type=str,
+                nargs="+",
+                help="Input collection(s) to search for coadd exposures.",
+                required=True,
+            )
+            sub.add_argument(
+                "-o",
+                "--hpix_build_order",
+                type=int,
+                default=1,
+                help="HEALPix order to segment sky for building quantum graph files.",
+            )
+            sub.add_argument(
+                "-w",
+                "--where",
+                type=str,
+                default=None,
+                help="Data ID expression used when querying for input coadd datasets.",
+            )
+
+        parser_build.add_argument(
+            "-q",
+            "--save-qgraph",
+            type=str,
+            help="Output filename for QuantumGraph.",
+            required=True,
+        )
+        parser_build.add_argument(
+            "-P",
+            "--pixels",
+            type=int,
+            nargs="+",
+            help="Pixels at --hpix_build_order to generate quantum graph.",
+            required=True,
+        )
+
+        args = parser.parse_args(argv)
+
+        if args.subparser_name is None:
+            parser.print_help()
+            sys.exit(1)
+
+        pipeline = pipeBase.Pipeline.from_uri(args.pipeline)
+        expanded_pipeline = list(pipeline.toExpandedPipeline())
+
+        if len(expanded_pipeline) != 1:
+            raise RuntimeError(f"Pipeline file {args.pipeline} may only contain one task.")
+
+        (task_def,) = expanded_pipeline
+
+        butler = Butler(args.butler_config, collections=args.input)
+
+        if args.subparser_name == 'segment':
+            # Do the segmentation
+            hpix_pixelization = HealpixPixelization(level=args.hpix_build_order)
+            conn = task_def.config.connections
+            dataset = conn.coadd_exposure_handles.format(coaddName=conn.coaddName)
+            data_ids = set(butler.registry.queryDataIds("tract", datasets=dataset).expanded())
+            region_pixels = []
+            for data_id in data_ids:
+                region = data_id.region
+                pixel_range = hpix_pixelization.envelope(region)
+                for r in pixel_range.ranges():
+                    region_pixels.extend(range(r[0], r[1]))
```

Comment:
Well, it doesn't want the input to be sorted (it can handle that), but it returns sorted data (which we want).  So we could use a set and then sort, or use a list and unique ... this isn't going to be a big list anyway, so I'm just going to leave it.